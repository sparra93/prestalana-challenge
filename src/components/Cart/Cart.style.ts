import styled from 'styled-components';

export const CartItemContainer = styled.div`
  height: 100vh;
`;
export const CartItem = styled.div`
  display: flex;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid #ccc;
  background-color: #f5f5f5;
  width: 300px;
  &:hover {
    background-color: #eee;
  }
`;

export const ItemImage = styled.img`
  width: 50px;
  height: 50px;
  object-fit: cover;
  margin-right: 1rem;
`;

export const ItemDetails = styled.div`
  flex-grow: 1;
`;

export const ItemName = styled.span`
  font-weight: bold;
  margin-bottom: 5px;
  color: black;
`;

export const ItemProperty = styled.p`
  color: black;
`;
