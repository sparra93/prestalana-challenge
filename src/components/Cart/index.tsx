import React from 'react';
import { useSelector } from 'react-redux';

import { selectCart } from '@/redux/features/productSlice';

import {
  CartItem,
  CartItemContainer,
  ItemDetails,
  ItemImage,
  ItemName,
  ItemProperty,
} from './Cart.style';

const Cart: React.FC = () => {
  const cartProducts = useSelector(selectCart);

  return (
    <CartItemContainer>
      {cartProducts.map((item) => (
        <CartItem key={item.id}>
          <ItemImage src="/assets/icons/shopping-bag.png" alt={item.name} />
          <ItemDetails>
            <ItemName>{item.name.toUpperCase()}</ItemName>
            <ItemProperty>
              <span style={{ fontWeight: 'bold' }}>Year:</span> {item.year}
            </ItemProperty>
            <ItemProperty>
              <span style={{ fontWeight: 'bold' }}>Pantone:</span>{' '}
              {item.pantone_value}
            </ItemProperty>
          </ItemDetails>
        </CartItem>
      ))}
    </CartItemContainer>
  );
};

export default Cart;
