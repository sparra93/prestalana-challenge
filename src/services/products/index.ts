import { ENDPOINT_URL, getUrl } from '@/shared/helpers/path';

export async function fetchProducts(): Promise<Response> {
  const getAllProductUrl = getUrl(ENDPOINT_URL.PRODUCT.ALL);

  const response = await fetch(getAllProductUrl, {
    method: 'GET',
    headers: { 'Content-Type': 'application/json' },
  });
  return response;
}
