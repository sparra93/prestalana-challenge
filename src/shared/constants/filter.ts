import type { IFilterState } from '@/shared/Interfaces/Filter.interface';

export const filterSliceInitialState: IFilterState = {
  yearFilter: [],
  colorFilter: [],
  pantoneFilter: [],
  nameFilter: '',
};
