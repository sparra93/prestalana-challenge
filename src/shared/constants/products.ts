import type { IProductsState } from '../Interfaces/Product.interface';

export const productSliceInitialState: IProductsState = {
  products: [],
  cart: [],
  favorites: [],
  error: false,
  loading: false,
  loaded: false,
  cartCountTotal: 0,
};
