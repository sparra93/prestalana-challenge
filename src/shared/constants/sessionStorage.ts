export const TOKEN_NAME = 'token';
export const EMAIL_NAME = 'email';
export const FAVORITES_NAME = 'FAVORITES';
