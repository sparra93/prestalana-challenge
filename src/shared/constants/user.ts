import type { IUserSliceInitialState } from '../Interfaces/User.interface';

export const userSliceInitialState: IUserSliceInitialState = {
  userInfo: null,
  error: false,
  loading: false,
  loaded: false,
  favorites: [],
};
