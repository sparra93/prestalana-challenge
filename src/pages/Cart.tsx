import type { NextPage } from 'next';
import React from 'react';

import Cart from '@/components/Cart';
import withAuth from '@/context/withAuth';

interface CartPageProps {}

const CartPage: NextPage<CartPageProps> = () => {
  return <Cart />;
};

export default withAuth(CartPage);
