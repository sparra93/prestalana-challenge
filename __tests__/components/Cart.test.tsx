import React from 'react';

import Cart from '@/components/Cart';
import { productSliceInitialState } from '@/shared/constants/products';

describe('Cart Component', () => {
  const customState = {
    products: {
      ...productSliceInitialState,
      cart: [
        {
          id: 1,
          name: 'cerulean',
          year: 2000,
          color: '#98B2D1',
          pantone_value: '15-4020',
          quantity: 1,
        },
        {
          id: 2,
          name: 'fuchsia rose',
          year: 2001,
          color: '#C74375',
          pantone_value: '17-2031',
          quantity: 1,
        },
      ],
    },
  };

  it('matches snapshot', () => {
    const { asFragment } = renderWithProviders(<Cart />, customState);

    expect(asFragment()).toMatchSnapshot();
  });
});
