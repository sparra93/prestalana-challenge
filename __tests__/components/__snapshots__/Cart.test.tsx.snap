// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ebf233ff-0 cSFOky"
  >
    <div
      class="sc-ebf233ff-1 kGXRWM"
    >
      <img
        alt="cerulean"
        class="sc-ebf233ff-2 ffgBwu"
        src="/assets/icons/shopping-bag.png"
      />
      <div
        class="sc-ebf233ff-3 fqKFUT"
      >
        <span
          class="sc-ebf233ff-4 eSCPjq"
        >
          CERULEAN
        </span>
        <p
          class="sc-ebf233ff-5 furSap"
        >
          <span
            style="font-weight: bold;"
          >
            Year:
          </span>
           2000
        </p>
        <p
          class="sc-ebf233ff-5 furSap"
        >
          <span
            style="font-weight: bold;"
          >
            Pantone:
          </span>
           15-4020
        </p>
      </div>
    </div>
    <div
      class="sc-ebf233ff-1 kGXRWM"
    >
      <img
        alt="fuchsia rose"
        class="sc-ebf233ff-2 ffgBwu"
        src="/assets/icons/shopping-bag.png"
      />
      <div
        class="sc-ebf233ff-3 fqKFUT"
      >
        <span
          class="sc-ebf233ff-4 eSCPjq"
        >
          FUCHSIA ROSE
        </span>
        <p
          class="sc-ebf233ff-5 furSap"
        >
          <span
            style="font-weight: bold;"
          >
            Year:
          </span>
           2001
        </p>
        <p
          class="sc-ebf233ff-5 furSap"
        >
          <span
            style="font-weight: bold;"
          >
            Pantone:
          </span>
           17-2031
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
